nixos/modules/system/boot/systemd/shutdown.nix

```diff
@@ -1,31 +1,57 @@
-{ config, lib, ... }: let
+{ config, lib, utils, pkgs, ... }: let
 
-  cfg = config.boot.systemd.shutdown;
+  cfg = config.systemd.shutdownRamfs;
+
+  ramfsContents = let
+    storePaths = map (p: "${p}\n") cfg.storePaths;
+    contents = lib.mapAttrsToList (_: v: "${v.source}\n${v.target}") (lib.filterAttrs (_: v: v.enable) cfg.contents);
+  in pkgs.writeText "shutdown-ramfs-contents" (lib.concatStringsSep "\n" (storePaths ++ contents));
 
 in {
-  options.boot.systemd.shutdown = {
+  options.systemd.shutdownRamfs = {
     enable = lib.mkEnableOption "pivoting back to an initramfs for shutdown" // { default = true; };
+    contents = lib.mkOption {
+      description = "Set of files that have to be linked into the shutdown ramfs";
+      example = lib.literalExpression ''
+        {
+          "/lib/systemd/system-shutdown/zpool-export".source = writeShellScript "zpool" "exec ''${zfs}/bin/zpool export -a"
```

Comment:
```suggestion
          "/lib/systemd/system-shutdown/zpool-export".source = writeShellScript "zpool" {} "exec ''${zfs}/bin/zpool export -a"
```

Reply:
Hm? `writeShellScript` only takes two arguments.

Reply:
Sorry, I mistook it with `runCommand` :(